homeassistant/components/fyta/__init__.py

```diff
@@ -0,0 +1,82 @@
+"""Initialization of FYTA integration."""
+from __future__ import annotations
+
+from datetime import datetime
+import logging
+
+from fyta_cli.fyta_connector import FytaConnector
+from fyta_cli.fyta_exceptions import (
+    FytaAuthentificationError,
+    FytaConnectionError,
+    FytaPasswordError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.device_registry import DeviceEntry
+
+from .const import DOMAIN
+from .coordinator import FytaCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = (Platform.SENSOR,)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the Fyta integration."""
+
+    username = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+    access_token = entry.data.get("access_token", "")
+    expiration = (
+        entry.data["expiration"] if "expiration" in entry.data else datetime.now()
+    )
+
+    fyta = FytaConnector(username, password, access_token, expiration)
+
+    coordinator = FytaCoordinator(hass, fyta, entry)
```

Comment:
Please refresh before setting to Hass.data

Reply:
what do you mean by refresh?

Reply:
`async_entry_first_refresh()` or something along those lines. Am on mobile so I can't check